src/group_theory/solvable.lean

```diff
@@ -331,3 +334,131 @@ lemma is_simple_group.comm_iff_is_solvable :
 end⟩
 
 end is_simple_group
+
+section perm_not_solvable
+
+lemma not_solvable_of_mem_derived_series {g : G} (h1 : g ≠ 1)
+  (h2 : ∀ n : ℕ, g ∈ derived_series G n) : ¬ is_solvable G :=
+mt (is_solvable_def _).mp (not_exists_of_forall_not
+  (λ n h, h1 (subgroup.mem_bot.mp ((congr_arg (has_mem.mem g) h).mp (h2 n)))))
+
+namespace S5_not_solvable
+
+/-- A type with 5 terms -/
+inductive weekday : Type
+| monday : weekday
+| tuesday : weekday
+| wednesday : weekday
+| thursday : weekday
+| friday : weekday
+
+namespace weekday
+
+instance : inhabited weekday := ⟨monday⟩
+
+instance : decidable_eq weekday :=
+begin
+  rintros (_ | _) (_ | _),
+  any_goals { exact is_true rfl },
+  all_goals { exact is_false (by trivial) },
+end
+
+/-- A 3-cycle -/
+def g1 : weekday → weekday
+| monday := monday
+| tuesday := tuesday
+| wednesday := thursday
+| thursday := friday
+| friday := wednesday
+
+/-- A 3-cycle -/
+def g2 : weekday → weekday
+| monday := monday
+| tuesday := tuesday
+| wednesday := friday
+| thursday := wednesday
+| friday := thursday
+
+/-- A (2,2)-cycle -/
+def g3 : weekday → weekday
+| monday := thursday
+| tuesday := friday
+| wednesday := wednesday
+| thursday := monday
+| friday := tuesday
+
+/-- A 3-cycle -/
+def g4 : weekday → weekday
+| monday := wednesday
+| tuesday := tuesday
+| wednesday := friday
+| thursday := thursday
+| friday := monday
+
+/-- A 3-cycle -/
+def g5 : weekday → weekday
+| monday := friday
+| tuesday := tuesday
+| wednesday := monday
+| thursday := thursday
+| friday := wednesday
+
+/-- A 3-cycle -/
+def σ1 : weekday ≃ weekday :=
+{ to_fun := g1,
+  inv_fun := g2,
+  left_inv := λ x, by { cases x, all_goals { refl } },
+  right_inv := λ x, by { cases x, all_goals { refl } } }
+
+/-- A (2,2)-cycle -/
+def σ2 : weekday ≃ weekday :=
+{ to_fun := g3,
+  inv_fun := g3,
+  left_inv := λ x, by { cases x, all_goals { refl } },
+  right_inv := λ x, by { cases x, all_goals { refl } } }
+
+/-- A 3-cycle -/
+def σ3 : weekday ≃ weekday :=
+{ to_fun := g4,
+  inv_fun := g5,
+  left_inv := λ x, by { cases x, all_goals { refl } },
+  right_inv := λ x, by { cases x, all_goals { refl } } }
+
+lemma mem_derived_series (n : ℕ) : σ1 ∈ derived_series (equiv.perm weekday) n :=
+begin
+  induction n with n ih,
+  { exact mem_top σ1 },
+  rw (show σ1 = σ3 * ((σ2 * σ1 * σ2) * σ1 * (σ2 * σ1 * σ2⁻¹)⁻¹ * σ1⁻¹) * σ3⁻¹,
+      by { ext, cases x, all_goals { refl } }),
+  exact (derived_series_normal _ _).conj_mem _
+    (general_commutator_containment _ _ ((derived_series_normal _ _).conj_mem _ ih _) ih) _,
+end
+
+lemma not_solvable : ¬ is_solvable (equiv.perm weekday) :=
+ not_solvable_of_mem_derived_series (mt equiv.ext_iff.mp
+  (not_forall_of_exists_not ⟨wednesday, by trivial⟩)) mem_derived_series
+
+lemma card : cardinal.mk weekday = ↑5 :=
+begin
+  letI : fintype weekday := fintype.mk
+    (finset.mk ↑[monday, tuesday, wednesday, thursday, friday] dec_trivial)
+    (by { rintros (_ | _), all_goals { dec_trivial } }),
```

Comment:
I think you should just make this a global instance rather than constructing it locally.